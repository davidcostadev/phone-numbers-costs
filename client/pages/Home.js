import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'recompose';
import { numbersType, paginatorType } from '../types';
import Numbers from '../components/Numbers';
import SelectPerPage from '../components/SelectPerPage';
import Topbar from '../components/Topbar';
import Bottombar from '../components/Bottombar';
import Paginator from '../components/Paginator';
import Page from '../components/Loading';

import withNumbers from '../connect/withNumbers';
import withPaginator from '../connect/withPaginator';
import setPerPage from '../connect/setPerPage';
import setPage from '../connect/setPage';
import fetchNumbers from '../connect/fetchNumbers';

const App = ({
  numbers,
  onChangePerPage,
  onClickPage,
  paginator,
  loading,
}) => (
  <Page loading={loading}>
    <Topbar title="Phone and Costs">
      <SelectPerPage paginator={paginator} onChangePerPage={onChangePerPage} />
    </Topbar>
    <main className="wrap">
      <Numbers numbers={numbers} />
    </main>
    <Bottombar>
      <Paginator paginator={paginator} onClickPage={onClickPage} />
    </Bottombar>
  </Page>
);

App.propTypes = {
  numbers: numbersType.isRequired,
  paginator: paginatorType,
  onChangePerPage: PropTypes.func.isRequired,
  onClickPage: PropTypes.func.isRequired,
  loading: PropTypes.bool.isRequired,
};

App.defaultProps = {
  paginator: null,
};


export default compose(
  withNumbers,
  withPaginator,
  setPerPage,
  setPage,
  fetchNumbers,
)(App);
