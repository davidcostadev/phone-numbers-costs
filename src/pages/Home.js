import React from 'react';
import PropTypes from 'prop-types';
import { compose } from 'recompose';
import { numbersType, paginatorType } from '../types';
import Numbers from '../components/Numbers';
import SelectPerPage from '../components/SelectPerPage';
import Topbar from '../components/Topbar';
import Bottombar from '../components/Bottombar';
import Paginator from '../components/Paginator';

import withNumbers from '../connect/withNumbers';
import withPaginator from '../connect/withPaginator';
import setPerPage from '../connect/setPerPage';
import setPage from '../connect/setPage';
import fetchNumbers from '../connect/fetchNumbers';

const App = ({
  numbers,
  onChangePerPage,
  onClickPage,
  paginator,
}) => (
  <div className="page">
    <Topbar title="Phone and Costs">
      <SelectPerPage paginator={paginator} onChangePerPage={onChangePerPage} />
    </Topbar>
    <main className="wrap">
      <Numbers numbers={numbers} />
    </main>
    <Bottombar>
      <Paginator paginator={paginator} onClickPage={onClickPage} />
    </Bottombar>
  </div>
);

App.propTypes = {
  numbers: numbersType.isRequired,
  paginator: paginatorType,
  onChangePerPage: PropTypes.func.isRequired,
  onClickPage: PropTypes.func.isRequired,
};

App.defaultProps = {
  paginator: null,
};


export default compose(
  withNumbers,
  withPaginator,
  setPerPage,
  setPage,
  fetchNumbers,
)(App);
