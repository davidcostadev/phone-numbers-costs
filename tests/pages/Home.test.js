import React from 'react';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { MemoryRouter, Route } from 'react-router-dom';
import { Provider } from 'react-redux';
import Home from '../../src/pages/Home';

const mockStore = configureMockStore([thunk]);

describe('Home Component', () => {
  let store;

  beforeEach(() => {
    store = mockStore({
      numbers: [],
    });
  });

  describe('Render', () => {
    it('default', () => {
      const tree = renderer.create((
        <MemoryRouter>
          <Provider store={store}>
            <Route render={props => (
              <Home {...props} />
            )}
            />
          </Provider>
        </MemoryRouter>
      )).toJSON();

      expect(tree).toMatchSnapshot();
    });
  });
});
