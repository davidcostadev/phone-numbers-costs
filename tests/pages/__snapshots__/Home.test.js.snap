// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Component Render default 1`] = `
<div
  className="page"
>
  <header
    className="topbar"
  >
    <h1
      className="topbar__title"
    >
      Phone and Costs
    </h1>
    <div
      className="topbar__right"
    />
  </header>
  <main
    className="wrap"
  >
    <div
      className="numbers__list"
    >
      <div
        className="numbers__empty"
      >
        no numbers found
      </div>
    </div>
  </main>
  <div
    className="bottombar"
  />
</div>
`;

exports[`Home Component Render with props 1`] = `
<div
  className="page"
>
  <header
    className="topbar"
  >
    <h1
      className="topbar__title"
    >
      Phone and Costs
    </h1>
    <div
      className="topbar__right"
    >
      <div
        className="select-per-page"
      >
        <select
          id="per-page"
          onChange={[Function]}
          value={100}
        >
          <option
            value={10}
          >
            10
          </option>
          <option
            value={50}
          >
            50
          </option>
          <option
            value={100}
          >
            100
          </option>
          <option
            value={250}
          >
            250
          </option>
          <option
            value={500}
          >
            500
          </option>
        </select>
      </div>
    </div>
  </header>
  <main
    className="wrap"
  >
    <div
      className="numbers__list"
    >
      <div
        className="numbers__item"
      >
        <div
          className="numbers__number"
        >
          555000000
        </div>
        <div
          className="numbers__costs"
        >
          1
        </div>
      </div>
      <div
        className="numbers__item"
      >
        <div
          className="numbers__number"
        >
          555000001
        </div>
        <div
          className="numbers__costs"
        >
          2
        </div>
      </div>
    </div>
  </main>
  <div
    className="bottombar"
  >
    <div
      className="paginator"
    >
      <button
        className="paginator__item paginator__item--disabled"
        disabled={true}
        id="paginator-previous"
        onClick={[Function]}
        type="button"
      >
        Previous
      </button>
      <button
        className="paginator__item paginator__item--active"
        disabled={false}
        id="paginator-1"
        onClick={[Function]}
        type="button"
      >
        1
      </button>
      <button
        className="paginator__item"
        disabled={false}
        id="paginator-2"
        onClick={[Function]}
        type="button"
      >
        2
      </button>
      <button
        className="paginator__item"
        disabled={false}
        id="paginator-3"
        onClick={[Function]}
        type="button"
      >
        3
      </button>
      <button
        className="paginator__item"
        disabled={false}
        id="paginator-next"
        onClick={[Function]}
        type="button"
      >
        Next
      </button>
    </div>
  </div>
</div>
`;
